### PR TITLE
[ALLUXIO-3115] Enable recoverLease for HDFS journal

### PR DESCRIPTION
**Pull request by gpang**
https://alluxio.atlassian.net/browse/ALLUXIO-3115

### CODE REVIEW

**Comment by gpang** [comment]
@aaudiber

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 1]
```diff
@@ -441,6 +447,25 @@ public InputStream open(String path, OpenOptions options) throws IOException {
...
+          try {
+            if (dfs.recoverLease(new Path(path))) {
+              LOG.warn("recoverLease-1 success for: {}", path);
+            } else {
+              // try one more time, after waiting
+              CommonUtils.sleepMs(5 * Constants.SECOND_MS);
+              if (dfs.recoverLease(new Path(path))) {
+                LOG.warn("recoverLease-2 success for: {}", path);
+              } else {
+                LOG.warn("recoverLease returned false for: {}", path);
```
Comment (aaudiber) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: can we specify what false indicates?

Comment (gpang) [reply underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: I updated the message.

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 2]
```diff
@@ -441,6 +447,29 @@ public InputStream open(String path, OpenOptions options) throws IOException {
...
         te = e;
+        if (options.getRecoverFailedOpen() && dfs != null && e.getMessage().toLowerCase()
+            .startsWith("cannot obtain block length for")) {
+          // This error can occur when an Alluxio journal file was not properly closed by Alluxio.
+          // In this scenario, the HDFS lease must be recovered in order for the file to be
+          // readable again. The 'recoverLease' API usually needs to be invoked multiple times
+          // to complete the lease recovery process.
+          try {
+            if (dfs.recoverLease(new Path(path))) {
+              LOG.warn("recoverLease-1 success for: {}", path);
```
Comment (tedyu) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: This should be logged at INFO or DEBUG

File: underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java [hunk 3]
```diff
@@ -441,6 +447,29 @@ public InputStream open(String path, OpenOptions options) throws IOException {
       } catch (IOException e) {
         LOG.warn("{} try to open {} : {}", retryPolicy.getRetryCount(), path, e.getMessage());
         te = e;
+        if (options.getRecoverFailedOpen() && dfs != null && e.getMessage().toLowerCase()
+            .startsWith("cannot obtain block length for")) {
```
Comment (tedyu) [inline underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java]: I searched hadoop 3.1 code base but didn't find where this exception message comes from

**Pull request merged**